src/op_analytics/datapipeline/chains/aggregator.py

```diff
@@ -0,0 +1,126 @@
+    # - Dune transaction data from dune_all_txs (Dune source, source_rank=4.5)
+    # - DefiLlama chain TVL data from daily_defillama_chain_tvl (DefiLlama source, source_rank=5)
+    # - Manual mappings CSV file for known corrections/overrides
+    # Each source loader should return standardized DataFrames with consistent columns
+    log.info("Step 1: Load Data - Not yet implemented")
+
+    # Step 2: Combine
+    # Intent: Concatenate all source DataFrames into a single unified dataset:
+    # - Union all source_dfs into all_sources_df using pd.concat()
+    # - Ensure consistent schema across all sources before concatenation
+    # - Log total records combined from all sources
+    # - This creates the raw comprehensive dataset for further processing
+    log.info("Step 2: Combine - Not yet implemented")
+
+    # Step 3: Preprocess
+    # Intent: Clean and standardize the combined data with key transformations:
+    # - Generate chain_key column (hash of chain_name for grouping similar names)
+    # - Standardize chain_id column (ensure consistent format, handle special cases)
+    # - Apply data type conversions and handle missing values
+    # - Normalize chain names, symbols for consistency
+    # - This prepares data for entity resolution and grouping
+    log.info("Step 3: Preprocess - Not yet implemented")
+
+    # Step 4: Entity Resolution
+    # Intent: Generate unified_key and apply manual mappings before grouping:
+    # - Create unified_key column combining chain_id, chain_key, and display_name logic
+    # - Apply manual mappings to correct data inconsistencies (e.g., chain_id overrides)
+    # - Handle special cases like 'molten' chain_id corrections based on source
+    # - This creates canonical identifiers for grouping duplicate entities
+    # - Manual mappings are applied BEFORE grouping to ensure correct entity resolution
+    log.info("Step 4: Entity Resolution - Not yet implemented")
+
+    # Step 5: Deduplication
+    # Intent: Group by unified_key and merge attributes using source prioritization:
+    # - Group records by unified_key (represents same chain entity)
+    # - Within each group, sort by source_rank (lower rank = higher priority)
+    # - Select primary record from highest priority source
+    # - Aggregate fields: min_dt_day (min), max_dt_day (max), data_sources (concat), all_chain_keys (concat)
+    # - Apply secondary deduplication using is_dupe logic for remaining duplicates
+    # - Results in one canonical record per unique chain
+    log.info("Step 5: Deduplication - Not yet implemented")
+
+    # Step 6: Enrichment
+    # Intent: Enhance deduplicated data with op_stack_chain_metadata enrichment:
+    # - Load op_stack_chain_metadata specifically for enrichment (not as competing source)
+    # - Left join with deduplicated data on chain_id (handle Celo -l2 suffix matching)
+    # - Coalesce/add fields: gas_token, da_layer, public_mainnet_launch_date, etc.
```

Comment:
yeah these "added fields" could either be generated from the top, or we have some way to easily join the grouped chains back to their original datasets, such that we can pull the appropriate fields.

Like:
Chain A has aliases B for Dataset 1, C for Dataset 2, D for Dataset 3, etc and we left join back on that